### ISSUE TITLE
Roberta for semantic similarity

### ISSUE THREAD

**Issue by subhamkhemka** [pytorch/fairseq#1066]
Hi 

Thanks for releasing Roberta.
I am trying to use Roberta for semantic similarity.
have 2 questions 

Can you validate my code to check if its able to correctly execute sentence-pair classification ?

I want to train  the roberta-mnli model on my own corpus, how do I do this?

Code : 
`
from pytorch_transformers import RobertaModel, RobertaTokenizer
from pytorch_transformers import RobertaForSequenceClassification, RobertaConfig

config = RobertaConfig.from_pretrained('roberta-large')
config.num_labels = len(list(label_to_ix.values()))
tokenizer = RobertaTokenizer.from_pretrained('roberta-large-mnli')
model = RobertaForSequenceClassification(config)
def prepare_features(seq_1,seq_2):
    aa=tokenizer.encode(seq_1)
    bb=tokenizer.encode(seq_2)
    zz=tokenizer.add_special_tokens_sentences_pair(aa,bb)
    input_ids=torch.tensor(zz)
    input_mask = [1] * len(zz)
    return torch.tensor(input_ids).unsqueeze(0), input_mask
class Intents(Dataset):
    def __init__(self, dataframe):
        self.len = len(dataframe)
        self.data = dataframe
        
    def __getitem__(self, index):
        utterance = self.data.q1[index]
        sent2 = self.data.q2[index]
        label = self.data.label[index]
        X, _  = prepare_features(utterance,sent2)
        y = label_to_ix[self.data.label[index]]
        return X, y
    
    def __len__(self):
        return self.len
train_size = 0.95
train_dataset=dataset.sample(frac=train_size,random_state=200).reset_index(drop=True)
test_dataset=dataset.drop(train_dataset.index).reset_index(drop=True)

training_set = Intents(train_dataset)
testing_set = Intents(test_dataset)
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
model = model.cuda()
# Parameters
params = {'batch_size': 1,
          'shuffle': True,
          'num_workers': 1}
training_loader = DataLoader(training_set, **params)
testing_loader = DataLoader(testing_set, **params)
loss_function = nn.CrossEntropyLoss()
learning_rate = 1e-05
optimizer = optim.Adam(params =  model.parameters(), lr=learning_rate)

max_epochs = 3
model = model.train()
for epoch in tqdm_notebook(range(max_epochs)):
    print("EPOCH -- {}".format(epoch))
    for i, (sent, label) in enumerate(training_loader):
        optimizer.zero_grad()
        sent = sent.squeeze(0)
        if torch.cuda.is_available():
          sent = sent.cuda()
          label = label.cuda()
        output = model.forward(sent)[0]
        _, predicted = torch.max(output, 1)
        
        loss = loss_function(output, label)
        loss.backward()
        optimizer.step()
        
        if i%100 == 0:
            correct = 0
            total = 0
            for sent, label in testing_loader:
                sent = sent.squeeze(0)
                if torch.cuda.is_available():
                  sent = sent.cuda()
                  label = label.cuda()
                output = model.forward(sent)[0]
                _, predicted = torch.max(output.data, 1)
                total += label.size(0)
                correct += (predicted.cpu() == label.cpu()).sum()
            accuracy = 100.00 * correct.numpy() / total
            print('Iteration: {}. Loss: {}. Accuracy: {}%'.format(i, loss.item(), accuracy))

def get_reply(msg,msg1):
  model.eval()
  input_msg, _ = prepare_features(msg,msg1)
  if torch.cuda.is_available():
    input_msg = input_msg.cuda()
  output = model(input_msg)

`

**Comment by Arjunsankarlal** [comment]
Is the code for training/finetuning the roBERTa model is released? I couldn't able to find the train.py file for finetuning on GLUE tasks, as mentioned in the README.glue.md.
